int32 id
int32 speed
int32 offset
int32 done_behaviour
---
int32 success